### PR TITLE
Milestone 6: Tests

### PR DESCRIPTION
**Pull request by medaminedev66**
- Write unit tests for the files `operate.js` and `calculate.js` using Jest.
- Create unit tests for all React components:
-- Use Jest snapshots to test the components.
-- Use React Testing Library to simulate user interaction.

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hello  

You've done pretty well so far!
But for a few issues, you are almost done with this. Hang in there!


## Required Changes ♻️
_Please check the comments under the review._


Happy coding, Cheers! 🚀🚀

Do feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether it is valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__tests__/Header.test.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import renderer from 'react-test-renderer';
+import { BrowserRouter as Router } from 'react-router-dom';
+import Header from '../components/Header';
+
+describe('test the Quote components', () => {
+  it('renders without errors', () => {
+    const root = document.createElement('div');
+    ReactDOM.render(
```
Comment (mikenath223) [inline src/__tests__/Header.test.js]: - [ ] Two snapshot tests are obsolete kindly check and remove them.

File: src/__tests__/operate.test.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+test('calculate the deduction of 2 numbers', () => {
+  expect(operate(10, 4, '-')).toBe('6');
+});
+
+test('calculate the deduction of 2 numbers', () => {
+  expect(operate(10, 4, '-')).toBe('6');
+});
+
+test('calculate the deduction of 2 numbers', () => {
+  expect(operate(2, 4, 'x')).toBe('8');
```
Comment (mikenath223) [inline src/__tests__/operate.test.js]: - [ ] Please add tests for other operators: `+`, `%` and also test for edge cases like division by zero.

File: src/__tests__/Header.test.js (hunk 1 above)
Comment (medaminedev66) [reply src/__tests__/Header.test.js]: thanks for reviewing our pull requet. I didn't get it, should I remove the snapshot files ?

File: src/__tests__/operate.test.js (hunk 2 above)
Comment (medaminedev66) [reply src/__tests__/operate.test.js]: Ok

**Review by dasileker — APPROVED** [review]
### Hi @medaminedev66, 

### Project Approved ✅

Approved


Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
